services/cargohold/test/integration/API.hs

```diff
@@ -131,6 +132,20 @@ testSimpleRoundtrip = do
       let utc' = parseTimeOrError False defaultTimeLocale rfc822DateFormat date' :: UTCTime
       liftIO $ assertBool "bad date" (utc' >= utc)
 
+testUnverifiedUserUpload :: TestM ()
+testUnverifiedUserUpload = do
+  let def = V3.defAssetSettings
+  let rets = [minBound ..]
+  let sets = def : map (\r -> def & V3.setAssetRetention ?~ r) rets
+  mapM_ simpleUpload sets
+  where
+    simpleUpload sets = do
+      uid <- liftIO $ Id <$> nextRandom
+      let bdy = (applicationText, "Hello World")
+      void $
+        uploadSimple (path "/assets/v3") uid sets bdy
+          <!! const 403 === statusCode
```

Comment:
All new tests should go into the new test framework.

Reply:
I've moved this over.